### PR TITLE
Feat: Evaluate missing languages

### PR DESCRIPTION
**Pull request by Samoed**
## Checklist
- [X] Run tests locally to make sure nothing is broken using `make test`. 
- [X] Run the formatter to format the code using `make lint`. 

Continue ideas from https://github.com/embeddings-benchmark/mteb/pull/1525

If a missing split is present in the existing results, return its results without running the evaluation.

### CODE REVIEW

File: mteb/evaluation/MTEB.py [hunk 1]
```diff
@@ -398,6 +401,7 @@ def run(
             output_folder: Folder where the results will be saved. Default to 'results'. Where it will save the results in the format:
                 `{output_folder}/{model_name}/{model_revision}/{task_name}.json`.
             eval_splits: List of splits to evaluate on. If None, the splits are taken from the task metadata.
+            eval_langs: List of langs to evaluate on. If None, the splits are taken from the task metadata.
```
Comment (KennethEnevoldsen) [inline mteb/evaluation/MTEB.py:404]: is this hf subset? - would not use lang (only for 3 letter iso codes)

File: mteb/evaluation/MTEB.py [hunk 2]
```diff
@@ -455,9 +459,19 @@ def run(
                 f"\n\n********************** Evaluating {task.metadata.name} **********************"
             )
 
+            task_eval_splits = (
+                eval_splits if eval_splits is not None else task.eval_splits
+            )
+            task_eval_langs = (
+                task.metadata.eval_langs
```
Comment (KennethEnevoldsen) [inline mteb/evaluation/MTEB.py:466]: `TaskMetadata.hf_subsets_to_langscripts` standardized everything to a dict I would use this throughout

File: mteb/evaluation/MTEB.py [hunk 3]
```diff
@@ -637,3 +675,64 @@ def get_last_evaluated_splits(self):
...
+                subsets_to_consider = []
+                for hf_subset, lang_list in task_eval_langs.items():
+                    # lang_list are like ["eng-Latn", "deu-Latn"]
+                    iso_langs = [l.split("-")[0] for l in lang_list]
+                    # If any requested language is present in this subset, consider it
+                    if any(run_lang in iso_langs for run_lang in eval_langs):
+                        subsets_to_consider.append(hf_subset)
+        else:
+            # Not multilingual, just one "default" subset
+            subsets_to_consider = ["default"]
```
Comment (KennethEnevoldsen) [inline mteb/evaluation/MTEB.py:707]: See `hf_subsets_to_langscripts` above

**Review by KennethEnevoldsen — APPROVED** [review]
Looks good here only a few minor things

**Comment by Samoed** [comment]
Now this PR not working correctly if no splits or languages passed

File: mteb/evaluation/MTEB.py (hunk 1 above)
Comment (Samoed) [reply mteb/evaluation/MTEB.py:404]: I want to check if hf subsets have these languages and eval only these languages. What are you suggesting?

**Comment by Samoed** [comment]
I have a small issue. When I run `mteb run -m intfloat/multilingual-e5-small -t Banking77Classification --overwrite` with existing results, it doesn’t do anything and I don't understand why.

```
INFO:mteb.cli:Running with parameters: Namespace(model='intfloat/multilingual-e5-small', task_types=None, categories=None, tasks=['Banking77Classification'], languages=None, benchmarks=None, device=None, output_folder='results', verbosity=2, co2_tracker=False, eval_splits=None, model_revision=None, batch_size=None, overwrite=True, save_predictions=False, func=<function run at 0x7fc7bbf42440>)
INFO:mteb.models.sentence_transformer_wrapper:Model prompts will be overwritten with {'query': 'query: ', 'passage': 'passage: '}
INFO:mteb.evaluation.MTEB:

## Evaluating 1 tasks:
─────────────────────────────────────────────────────────────────────── Selected tasks  ────────────────────────────────────────────────────────────────────────
Classification
    - Banking77Classification, s2s


INFO:mteb.evaluation.MTEB:

********************** Evaluating Banking77Classification **********************
```

File: mteb/evaluation/MTEB.py (hunk 1 above)
Comment (KennethEnevoldsen) [reply mteb/evaluation/MTEB.py:404]: Hmm I would use hf_subset here and leave language selection to `mteb.get_tasks(...)`

**Comment by isaac-chung** [comment]
> I have a small issue. When I run `mteb run -m intfloat/multilingual-e5-small -t Banking77Classification --overwrite` with existing results, it doesn’t do anything and I don't understand why.

This runs fine for me in your branch.

**Comment by Samoed** [comment]
Hm, intresting

File: tests/test_evaluation/test_split_evaluation.py [hunk 4]
```diff
@@ -89,3 +99,206 @@ def test_no_missing_splits(model, tasks, tmp_path):
...
+    results = evaluation.run(
+        model,
+        eval_splits=["val"],
+        output_folder=str(tmp_path / "partial_langs_partial_splits"),
+        verbosity=2,
+        eval_langs=["eng", "fra"],
+    )
+
+    last_evaluated_splits = evaluation.get_last_evaluated_splits()
+    assert set(last_evaluated_splits["MockMultilingualRetrievalTask"]) == {"val"}
```
Comment (isaac-chung) [inline tests/test_evaluation/test_split_evaluation.py:234]:
```suggestion
    assert len(last_evaluated_splits["MockMultilingualRetrievalTask"]) == 1
    assert set(last_evaluated_splits["MockMultilingualRetrievalTask"]) == {"val"}
```

File: tests/test_evaluation/test_split_evaluation.py [hunk 5]
```diff
@@ -89,3 +99,206 @@ def test_no_missing_splits(model, tasks, tmp_path):
...
+    )
+    last_evaluated_splits = evaluation.get_last_evaluated_splits()
+    assert len(last_evaluated_splits["MockMultilingualRetrievalTask"]) == 1
+    assert last_evaluated_splits["MockMultilingualRetrievalTask"] == ["test"]
+    assert sorted(results[0].languages) == ["eng", "fra"]
+    assert results[0].scores.keys() == {"test"}
+    assert len(results[0].scores["test"]) == 2
+
+
+def test_multilingual_multiple_splits_partial_langs_partial_splits(
```
Comment (isaac-chung) [inline tests/test_evaluation/test_split_evaluation.py:221]: I was confused by the naming, as all langs are run, but not all splits are.
```suggestion
def test_multilingual_one_missing_split_no_missing_lang(
```

File: tests/test_evaluation/test_split_evaluation.py [hunk 6]
```diff
@@ -89,3 +99,206 @@ def test_no_missing_splits(model, tasks, tmp_path):
...
+
+    last_evaluated_splits = evaluation.get_last_evaluated_splits()
+    assert set(last_evaluated_splits["MockMultilingualRetrievalTask"]) == {"test"}
+    assert sorted(results[0].languages) == ["eng", "fra"]
+    assert results[0].scores.keys() == {"test", "val"}
+    assert len(results[0].scores["test"]) == 2
+    assert len(results[0].scores["val"]) == 2
+
+
+def test_multilingual_multiple_splits_missing_only_one_language_in_one_split(
```
Comment (isaac-chung) [inline tests/test_evaluation/test_split_evaluation.py:256]:
```suggestion
def test_multilingual_one_missing_lang_in_one_split(
```

File: mteb/evaluation/MTEB.py [hunk 7]
```diff
@@ -469,55 +480,74 @@ def run(
...
+                    # If the whole split is missing, run all required subsets
+                    # If only some subsets are missing, run only those
+                    if info["whole_split_missing"]:
+                        if task_subsets is not None:
+                            subsets_to_run = info["missing_subsets"]
+                        else:
+                            subsets_to_run = ["default"]
+                    else:
+                        # partial subsets missing
+                        subsets_to_run = info["missing_subsets"]
```
Comment (isaac-chung) [inline mteb/evaluation/MTEB.py:533]: Can be simplified to reduce mental load.
```suggestion
                    subsets_to_run = info["missing_subsets"]
                    if info["whole_split_missing"] and task_subsets is None:
                        subsets_to_run = ["default"]
```

**Review by isaac-chung — COMMENTED** [review]
Looks good. Just a few suggestions :)

File: README.md [hunk 8]
```diff
@@ -209,6 +209,21 @@ Note that the public leaderboard uses the test splits for all datasets except MS
 
 </details>
 
+
+<details>
```
Comment (isaac-chung) [inline README.md:213]: I thought we wanted this:  https://github.com/embeddings-benchmark/mteb/pull/1584#discussion_r1884403225

We might want to remove this section from the README.

Comment (Samoed) [reply README.md:213]: This was done according to the comment. I'll change the subset name to something different to make it less confusing

File: README.md [hunk 9]
```diff
@@ -209,6 +209,21 @@ Note that the public leaderboard uses the test splits for all datasets except MS
...
+  <summary> Selecting evaluation subset </summary>
+
+### Selecting evaluation subset
+You can evaluate only on selected subsets. For example, if you want to evaluate only the `subset_name_to_run` subset of all tasks, do the following:
+
+```python
+evaluation.run(model, eval_subsets=["subset_name_to_run"])
```
Comment (KennethEnevoldsen) [inline README.md:220]: Might be nice to give an actual example with a dataset (linking to the dataset on huggingface)

File: README.md (hunk 8 above)
Comment (isaac-chung) [reply README.md:213]: I see, thanks.

**Pull request merged**